### PR TITLE
Clean up maintenance banner

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-3531](https://dfedigital.atlassian.net.mcas.ms/jira/software/projects/CPDLP/boards/87?assignee=712020%3A5b5839dc-e4d6-4df5-9dd6-627cc49fe345%2Cunassigned&selectedIssue=CPDLP-3531)

### Context

We want to fix and clean up the maintenance banner in ECF so that we can use it for the upcoming downtime when we transition the lead provider APIs to NPQ reg.

### Changes proposed in this pull request

- Clean up maintenance banner

Clean up the maintenance banner to match the latest design.

Fix broken `Dismiss` button; swap to a link to be consistent with the GOV.UK banner component usage.

Change route to GET so we can use it from a regular link in the GOV.UK components helpers (not ideal but allows us to use the component to render the banner).

Add missing test coverage.

### Guidance to review

Screenshot 2024-10-29 at 14 33 52

### CODE REVIEW

**Review by leandroalemao — APPROVED** [review]
Tested and looking good tks @ethax-ross 👍

**Review by mooktakim — APPROVED** [review]

File: app/components/banners/maintenance_component.rb [hunk 1]
```diff
@@ -2,73 +2,67 @@
...
-    DATE       = nil # Date.new(2023, 9, 14)
-    START_TIME = nil # Time.zone.local(2023, 9, 14, 18, 0, 0)
-    END_TIME   = nil # Time.zone.local(2023, 9, 14, 21, 0, 0)
-
-    def initialize(wide_container_view: false)
-      @wide_container_view = wide_container_view
-    end
+    HOUR_FORMAT = "%-l%P"
+    DAY_FORMAT = "%-d %B"
+    MAINTENANCE_WINDOW = Time.zone.local(2024, 11, 27, 19)..Time.zone.local(2024, 11, 27, 21)
```
Comment (cwrw) [inline app/components/banners/maintenance_component.rb:7]: can we change this to 10pm please

**Review by cwrw — COMMENTED** [review]
looks good, just the time comment :)

**Review by leandroalemao — APPROVED** [review]
Also could you please link up the ticket # to the PR? Thanks

**Review by cwrw — APPROVED** [review]

**Pull request merged**